### PR TITLE
feat(guild): add fetch_joined_channel method

### PR DESCRIPTION
**Pull request by DeeChael**
implemented: [channel-user/get-joined-channel](https://developer.kaiheila.cn/doc/http/channel-user#%E6%A0%B9%E6%8D%AE%E7%94%A8%E6%88%B7id%E5%92%8C%E6%9C%8D%E5%8A%A1%E5%99%A8id%E8%8E%B7%E5%8F%96%E7%94%A8%E6%88%B7%E6%89%80%E5%9C%A8%E8%AF%AD%E9%9F%B3%E9%A2%91%E9%81%93)

### CODE REVIEW

File: khl/guild.py [hunk 1]
```diff
@@ -161,6 +161,15 @@ async def list_user(self, channel: Channel) -> List[User]:
         users = await self.gate.exec_pagination_req(api.Guild.userList(guild_id=self.id, channel_id=channel.id))
         return [User(_gate_=self.gate, _lazy_loaded_=True, **i) for i in users]
 
+    async def fetch_joined_channel(self, user: User, page: int = 1, page_size: int = 50) -> List[PublicChannel]:
+        response = await self.gate.exec_req(api.ChannelUser.getJoinedChannel(page=page, page_size=page_size, guild_id=self.id, user_id=user.id))
```
Comment (TWT233) [inline khl/guild.py:165]: there is a native paged req helper: `gate.exec_pagination_req()` which may make this work easier

Comment (TWT233) [reply khl/guild.py:165]: FYI: `Guild.list_user()/fetch_roles()` is an usage example for `exec_pagination_req()`

**Review by TWT233 — COMMENTED** [review]
thanks for PR, but the code can be simplified with helpers

btw the commit message linter ci fails

**Review by TWT233 — CHANGES_REQUESTED** [review]
there may be a better approach with a higher level helper

**Comment by DeeChael** [comment]
> thanks for PR, but the code can be simplified with helpers
> 
> btw the commit message linter ci fails

我看之前的commits里有用过feature啊

**Comment by TWT233** [comment]
> > thanks for PR, but the code can be simplified with helpers
> > btw the commit message linter ci fails
> 
> 我看之前的commits里有用过feature啊 已经改好了，但不知道怎么commit了

应该用 feat，以前还没commit lint所以我没注意就合进去了吧（

**Comment by DeeChael** [comment]
> 我看好像没改？是不是被merge覆盖了

还没push呢 XD

**Comment by TWT233** [comment]
看起来是最开始的commit没过：
```
input: feature(guild): add fetch_joined_channel method
✖   type must be one of [build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test] [type-enum]
✖   found 1 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
```

**Comment by DeeChael** [comment]
> 看起来是最开始的commit没过：
> 
> ```
> input: feature(guild): add fetch_joined_channel method
> ✖   type must be one of [build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test] [type-enum]
> ✖   found 1 problems, 0 warnings
> ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
> ```

那怎么处理？

**Comment by TWT233** [comment]
> > 看起来是最开始的commit没过：
> > ```
> > input: feature(guild): add fetch_joined_channel method
> > ✖   type must be one of [build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test] [type-enum]
> > ✖   found 1 problems, 0 warnings
> > ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
> > ```
> 
> 那怎么处理？

这里是 `feature(guild): add fetch_joined_channel method` 中开头的 feature 应该换成 feat，我一般会用 rebase 来改 commit message（同时可以顺便 squash 下）

**Comment by DeeChael** [comment]
> > > 看起来是最开始的commit没过：
> > > ```
> > > input: feature(guild): add fetch_joined_channel method
> > > ✖   type must be one of [build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test] [type-enum]
> > > ✖   found 1 problems, 0 warnings
> > > ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
> > > ```
> > 
> > 
> > 那怎么处理？
> 
> 这里是 `feature(guild): add fetch_joined_channel method` 中开头的 feature 应该换成 feat，我一般会用 rebase 来改 commit message（同时可以顺便 squash 下）

整半天不会整，放弃了，我直接重开pull request得了

**Pull request closed without merging**